### DISCUSSION TITLE
Can we create multiple resources from one model ?

### DISCUSSION THREAD

**Discussion by chahal26** [filamentphp/filament#3052] [Help]
Can we create multiple resources from a single model? I want to create 2 different resources which will be differentiated based on the `user_type` field in the User model.

For example: If we have 2 types of users i.e. drivers and sellers. Both are saved in the same user's table differentiated by `user_type`. What would be the best way to create a separate resource for both?

**Comment by khanakia** [comment]
@danharrin i am looking for the same solution.
I want to create the Resource from the Same Model twice but with a different name. 

As i have 2 different user roles and they need to see different data and have different action on the same Model.

**Comment by ryangjchandler** [comment]
This is possible as long as the resources have different names / slugs. You'll need to change the base Eloquent query used so that it filters down correctly.

**Reply by khanakia** [reply]
Yes i know but how to change the name of the resource ? `php artisan make:filament-resource UserPaymentRequest` This command has no option to add Custom Resource Name and when i try to fire the same command again i get an error that resource already exists.

**Reply by danharrin** [reply]
Just rename the files you generate...

**Reply by khanakia** [reply]
@danharrin how do i apply different Policy as both Resouces has the same Model but i have different actions e.g for Edit Page for one resource only the parent of the user can edit the resource.

**Reply by danharrin** [reply]
Override the `can()` method on the resource to point to a different policy.

Please stop @ ing me all the time, I will see your question when I check my GitHub anyway. Other people are able to answer your questions too, not just me.

**Reply by khanakia** [reply]
Thanks. As the question was related to the previous question and conversation was happening with you already. 

Sure Won't ping you next time.

**Comment by shankhadevpadam** [comment]
Yes, I have created Post and Page resources using a single Post Model.
